src/Pos/Util/LogSafe.hs

```diff
@@ -7,37 +7,47 @@ module Pos.Util.LogSafe
        , logNoticeS
        , logWarningS
        , logErrorS
-       , HasSecureFormatting(..)
        ) where
 
 import           Control.Monad.Trans (MonadTrans)
-import           Formatting          (Format)
+import           Data.List           (isSuffixOf)
+import qualified Data.Text           as T
+import qualified System.Log.Logger   as L
 import           System.Wlog         (CanLog (..), HasLoggerName (..), Severity (..),
-                                      WithLogger)
+                                      convertSeverity, loggerName)
 import           Universum
 
 
 newtype SecureLogWrapped m a = SecureLogWrapped
     { getSecureLogWrapped :: m a
-    } deriving (Functor, Applicative, Monad, CanLog)
+    } deriving (Functor, Applicative, Monad, MonadIO)
 
 instance MonadTrans SecureLogWrapped where
     lift = SecureLogWrapped
 
+instance (MonadIO m) => CanLog (SecureLogWrapped m) where
+    dispatchMessage
+        (loggerName      -> name)
+        (convertSeverity -> prior)
+        msg =
+      let acceptable p
+              | "fileHandler" `isPrefixOf` p ||
+                "rollerHandler" `isPrefixOf` p
+              = not $ ".pub" `isSuffixOf` p
+              | otherwise = True
+      in liftIO $ L.logMCond name prior (T.unpack msg) acceptable
+
 instance (Monad m, HasLoggerName m) => HasLoggerName (SecureLogWrapped m) where
-    getLoggerName = (<> ".Secret") <$> SecureLogWrapped getLoggerName
+    getLoggerName = SecureLogWrapped getLoggerName
     modifyLoggerName foo (SecureLogWrapped m) =
         SecureLogWrapped (modifyLoggerName foo m)
 
 execSecureLogWrapped :: SecureLogWrapped m a -> m a
 execSecureLogWrapped (SecureLogWrapped act) = act
 
-class HasSecureFormatting a where
-    secureF :: Bool -> Format r (a -> r)
-
 -- | Shortcut for 'logMessage' to use according severity.
 logDebugS, logInfoS, logNoticeS, logWarningS, logErrorS
-    :: WithLogger m
+    :: (HasLoggerName m, MonadIO m)
```

Comment:
Most definitely. I barely managed to find a solution to make this happen (by forking `hslogger`). I don't know if it's easy to support secret logging in pure environment.